### PR TITLE
FHE-DKSAP Project Proposal

### PR DESCRIPTION
**Pull request by Mason-Mind**
draft the proposal for FHE-DKSAP project

### CODE REVIEW

File: projects/FHE-DKSAP.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+(What are the limitations and issues you may need to overcome?)
+
+The communication cost, computation cost and the complexity of implementing fully homomorphic encryption. Meanwhile, most FHE schemes can only support 64 bites. However, our stealth address is 256 bits. The problem lies in how to extend the length for FHE.
+
+## Goal of the project
+(What does success look like? Describe the end goal of the project, scope, state and impact for the project to be considered finished and successful.)
+
+- It protects privacy of stealth addresses by computing over ciphertext. 
+- Compared to DKSAP and BasedSAP, our design removes the risk of leakages on keys and personal information. 
+- Meanwhile, it can prevent quantum computing attacks as well.
```
Comment (JoshDavisLight) [inline projects/FHE-DKSAP.md:114]: I would love to see more here about what your personal success metrics look like, both for EPF and generally, and less about the actual benefit that FHE-DKSAP will have on the protocol, which you outline in the Motivation section. In what stage does this project need to be for you to consider it to be successful within EPF?

File: projects/FHE-DKSAP.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  
+## Collaborators
+
+## Fellows
+(Are there any fellows working with you on this project?)
+
```
Comment (JoshDavisLight) [inline projects/FHE-DKSAP.md:120]: Make sure to add your own name/link here, as well as anyone else you may be working with.

**Review by JoshDavisLight — CHANGES_REQUESTED** [review]

**Review by taxmeifyoucan — COMMENTED** [review]
Such a great proposal, I am very excited to see this!! 

One thing you could clarify in the specs/challenges part - how should wallets manage new kind of key pairs. Since you are introducing FHE-DKSAP with Paillier apart of regular secp256k1, Eth wallet would need to implement it and securely store another pair of keys. Maybe it's too much ahead, I am just wondering if there is anything you can discuss further on handling FHE keys

**Comment by Mason-Mind** [comment]
@taxmeifyoucan very inspiring comment. I have added some initial thought in the implementation part. But it will need more exploring during the implementation.

**Pull request merged**